Muon documentation update

AnthonyLim23 opened the pull request:
I have updated the muon documentation to reflect the new UI's in muon analysis (see #19545 and #19454) 

**To test:**

<!-- Instructions for testing. -->
Check that it renders correctly. 
Fixes #19585 

<!-- RELEASE NOTES
Either edit the file in docs/source/release/... and it will be in your pull request or state
*Does not need to be in the release notes.*
-->
*Does not need to be in the release notes.
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

DTasev commented on docs/source/interfaces/Muon_Analysis.rst:
```diff
@@ -306,36 +323,34 @@ This button will open the "Edit local parameter values" dialog, which offers gre
+^^^^^^^^^^
+
+The data table allows the user to modify the selected data for the fitting. This includes the start and end times, which can also
+be updated by dragging the blue dashed lines in the plot. The "Groups/Pairs to fit" box provides a drop-down menu with three options (all groups, all pairs and custom". 
```
I think there's a `"` at the end that should be a `)`

AnthonyLim23 replied:
you are right. Thank you

DTasev commented on docs/source/interfaces/Muon_Analysis.rst:
```diff
@@ -372,12 +387,27 @@ Examples/Use cases
-Options
-^^^^^^^
-The bottom of the tab contains selected fit options that can be adjusted, just as elsewhere in Mantid.
+Settings
```
I think Options was less confusing, I was looking for the mentioned options in the `Settings` tab. Maybe a longer name, something like:
- Data Analysis Options
- Additional Options

would be more clear, because there is already a Settings tab

AnthonyLim23 commented:
I have fixed those typos. Thank you for spotting them. I have also fixed another typo that was found in DynamicKuboToyabe (the upper limit of the integration should be t and not tau).

DTasev commented:
Looks good with the changes. I'm happy to :shipit:
